source/docs/casper/resources/tutorials/advanced/cross-contract.md

```diff
@@ -0,0 +1,538 @@
+import useBaseUrl from '@docusaurus/useBaseUrl';
+
+# Cross Contract Communication
+
+Level of the tutorial: `Advanced`
+
+Time to complete the tutorial: `45 - 60 Minutes`
+
+Important topics to read before going further:
+
+- [getting started](../../../developers/writing-onchain-code/getting-started.md)
+- [writing on-chain code](../../../developers/writing-onchain-code/simple-contract.md)
+- [calling contracts](../../../developers/writing-onchain-code/calling-contracts.md)
+
+This will assure that you have needed tools and base knowledge to complete this tutorial. If not everything was clear while reading these sections, don't worry as we will be doing a full walkthrough step by step.
+
+## Outline of the tutorial {#outline}
+
+This tutorial covers all possible variations of cross contract communication. While it not necessary when writing a simple contract which doesn't have to interact with other deployed contracts it is nontheless important to understand the ways to achieve this feat.
+
+On a very high level a smart contract behaves like a class in object oriented programming. This would suggest, that the principles for object oriented programming also apply for smart contract development. What interests us the most is the responsibility separation of classes. On a conceptual level for very simple functionalities there is no need to separate the logic between more smart contracts. It is advised to think about it and preplan the development. A smart contract on Casper Network can be always updated, but for more complex scenarios there is a need to fully understand how the project can be segrageted into smaller units of logical work.
+
+To better visualise this behaviour let's shortly look how the uniswap v2 protocol is built.
+
+Uniswap v2 protocol consists of multiple smart contracts which govern a unified blockchain application and each contract serves a different purpose.
+The contracts are as follows:
+- Factory
+- Pair
+- Pair (ERC20)
+- Library
+- Router01
+- Router02
+
+Even considering the names of the smart contracts we can see that each smart contract should have a different function.
+The Factory contract is generally used to create a token pair. It throws an event that a pair has been created and allows the user to read the pair created. The most important to notice is that the generation of a token pair actually creates a contract of type Pair under a new address hash.
+The Pair smart contract is used to perform operations like mint or burn on a created pair of tokens.
+
+The principle of responsibility separation is blockchain agnostic so analizing multi-contract applications will benefit every smart contract developer regardless of the technology used.
+Having this in mind we will be building two contracts which reference each other in some shape or form. We will look at how the keys are deployed in the contracts context and how we can pass the contract hash into a deployed contract so another contract can be called.
```

Comment:
"analizing multi.contract applications" ... poor applications. Please run a spell-checker on your work. There's a bunch of spelling errors through this document.

Reply:
Rewritten this part accordingly, spelling etc. was checked